analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/InitializationVectorShouldBeRandom.cs

```diff
@@ -0,0 +1,248 @@
+            {
+                programState = programPoint.CurrentInstruction switch
+                {
+                    ObjectCreationExpressionSyntax objectCreation => ObjectCreationPostProcess(objectCreation, programState),
+                    InvocationExpressionSyntax invocation => InvocationExpressionPostProcess(invocation, programState),
+                    ArrayCreationExpressionSyntax arrayCreation => ArrayCreationPostProcess(arrayCreation, programState),
+                    AssignmentExpressionSyntax assignment => AssignmentExpressionPostProcess(assignment, programState),
+                    _ => programState
+                };
+
+                return base.PostProcessInstruction(programPoint, programState);
+            }
+
+            public override ProgramState PreProcessInstruction(ProgramPoint programPoint, ProgramState programState)
+            {
+                if (programPoint.CurrentInstruction is InvocationExpressionSyntax invocation)
+                {
+                    programState = InvocationExpressionPreProcess(invocation, programState);
+                }
+
+                return base.PreProcessInstruction(programPoint, programState);
+            }
+
+            private ProgramState ObjectCreationPostProcess(ObjectCreationExpressionSyntax objectCreation, ProgramState programState)
+            {
+                if (semanticModel.GetSymbolInfo(objectCreation).Symbol is {} symbol
+                    && symbol.ContainingType is {} symbolType
+                    && symbolType.IsAny(VulnerableTypes))
+                {
+                    var symbolicValue = programState.PeekValue();
+                    programState = programState.SetConstraint(symbolicValue, IVInitializationSymbolicValueConstraint.NotInitialized);
+                    programState = programState.SetConstraint(symbolicValue, KeyInitializationSymbolicValueConstraint.NotInitialized);
+                }
+
+                return programState;
+            }
+
+            private ProgramState InvocationExpressionPostProcess(InvocationExpressionSyntax invocation, ProgramState programState)
+            {
+                if (IsCreateMethod(invocation, semanticModel))
+                {
+                    var symbolicValue = programState.PeekValue();
+                    programState = programState.SetConstraint(symbolicValue, IVInitializationSymbolicValueConstraint.NotInitialized);
+                    programState = programState.SetConstraint(symbolicValue, KeyInitializationSymbolicValueConstraint.NotInitialized);
+                }
+                else if (IsGenerateKeyMethod(invocation, semanticModel))
+                {
+                    var symbolicValue = GetSymbolicValue(invocation, programState);
+                    programState = programState.SetConstraint(symbolicValue, KeyInitializationSymbolicValueConstraint.Initialized);
+                }
```

Comment:
Why does this rule & PR cares about Key? It should be interested in IV only. Raising on Key sounds like a different RSPEC with different message (that we can easily implement now, not sure if we have it).

Reply:
Right. It shouldn't.